eng/SourceBuild.props

```
@@ -15,23 +15,36 @@
     <!-- Split e.g. 'fedora.33-x64' into 'fedora.33' and 'x64'. -->
     <_targetRidPlatformIndex>$(TargetRid.LastIndexOf('-'))</_targetRidPlatformIndex>
     <TargetArch>$(TargetRid.Substring($(_targetRidPlatformIndex)).TrimStart('-'))</TargetArch>
+    <TargetOS>$(TargetRid.Substring(0, $(_targetRidPlatformIndex)).TrimStart('-'))</TargetOS>
```

Do we expect to get invalid RIDs here with a prefixing `-`? If not, let's remove the `TrimStart`.

```suggestion
    <TargetOS>$(TargetRid.Substring(0, $(_targetRidPlatformIndex)))</TargetOS>
```